Detour/Source/DetourNavMeshQuery.cpp

```diff
@@ -998,106 +1106,207 @@ dtStatus dtNavMeshQuery::queryPolygons(const float* center, const float* halfExt
 			}
 		}
 	}
-	
+
 	return DT_SUCCESS;
 }
-
 /// @par
 ///
 /// If the end polygon cannot be reached through the navigation graph,
 /// the last polygon in the path will be the nearest the end polygon.
 ///
-/// If the path array is to small to hold the full result, it will be filled as 
+/// If the path array is to small to hold the full result, it will be filled as
 /// far as possible from the start polygon toward the end polygon.
 ///
-/// The start and end positions are used to calculate traversal costs. 
+/// The start and end positions are used to calculate traversal costs.
 /// (The y-values impact the result.)
 ///
 dtStatus dtNavMeshQuery::findPath(dtPolyRef startRef, dtPolyRef endRef,
 								  const float* startPos, const float* endPos,
 								  const dtQueryFilter* filter,
-								  dtPolyRef* path, int* pathCount, const int maxPath) const
+								  dtPolyRef* path, int* pathCount, const int maxPath,
+								  dtQueryPathState* state /* = 0 */) const
+{
+	return findMultiStartPath(1, &startRef, endRef, startPos, endPos, filter,
+			path, pathCount, maxPath, state);
+}
+
+/// @par
+///
+/// If the end polygon cannot be reached through the navigation graph,
+/// the last polygon in the path will be the nearest the end polygon.
+///
+/// If the path array is to small to hold the full result, it will be filled as
+/// far as possible from the start polygon toward the end polygon.
+///
+/// The start and end positions are used to calculate traversal costs.
+/// (The y-values impact the result.)
+///
+dtStatus dtNavMeshQuery::findMultiStartPath(const int numStarts,
+                                  const dtPolyRef* startRefs,
+		                          dtPolyRef endRef,
+								  const float* startPoses,
+								  const float* endPos,
+								  const dtQueryFilter* filter,
+								  dtPolyRef* path, int* pathCount,
+								  const int maxPath,
+								  dtQueryPathState* state /* = 0 */) const
```

Comment:
Is terminology accurate here? Would we want to call it `cache`or `cache_state` or `precalc_state`. Something that distinguish it from agent or position.